### ISSUE TITLE
Update flow definitions for 0.53 format

### ISSUE THREAD

**Issue by chollier** [erikras/redux-form#3345]
### **bug report** 

With the updated version of Flow (0.53), the React definitions have changed : https://github.com/facebook/flow/releases/tag/v0.53.0

### What is the current behavior?
When using the package's definition it now fails

### What is the expected behavior?
It wouldn't fail on redux-forms definitions

### Other information
I assume you should maybe release a new version with updated flow defs. Check out https://www.npmjs.com/package/flow-upgrade as is can automate some of the update

**Comment by h-michael** [comment]
@erikras 
React type definition was breaking change on flow v0.53.0.
Redux-form type definition is at this repository and it's flow version is locked on package.json.
There are both using flow lower v0.53.0 and upper v0.53.0 .
I think that we move  type definition to 'flow-typed', because of corresponding to a plurality of versions flow.

**Comment by SpearThruster** [comment]
There are some issues in flow in v0.53 that were immediately fixed in 0.54... so we need to target 0.54 atm and maybe rename this issue accordingly.

**Comment by ghostd** [comment]
Hi,

I started to update the flow definitions to flow@0.54 (a first pass with flow-upgrade, then some manually fixes/updates). The main issue i didnt yet fix is the type of the `saveRef` functions (and all `ref` fields usage). 

Let's take `ConnectedField`; according to the [documentation](https://flow.org/en/docs/react/types/#toc-react-ref), the type of `saveRef` should be:
```
saveRef = (ref: ?React.ElementRef<*>) => (this.ref = ref)
```
That means the `ref` declaration should be:
```
ref: ?React.ElementRef
```
That implies that all usage of `ref` will need to check null and `getRenderedComponent()` could return null (and so on).

Am i wrong with the doc ? Any ideas?

**Comment by SpearThruster** [comment]
@ghostd mind sharing your current work somewhere? I will gladly help.

**Comment by h-michael** [comment]
PR: #3378

**Comment by erikras** [comment]
I'd like to wait for the Travis problem to be fixed before merging #3378.

**Comment by erikras** [comment]
Published fix in [`v7.0.4`](https://github.com/erikras/redux-form/releases/tag/v7.0.4).
